tests/src/tests/command_test_utils.go

```diff
@@ -0,0 +1,258 @@
+	command := schema.Command{}
+	command.Id = GetRandomUniqueString(8, true)
+	LogInfoMessage(fmt.Sprintf("command Id: %s", command.Id))
+
+	if commandType == schema.ExecCommandType {
+		command.Exec = createExecCommand()
+	} else if commandType == schema.CompositeCommandType {
+		command.Composite = createCompositeCommand()
+	}
+	return command
+}
+
+// Update the values of a specified command
+func (devfile *TestDevfile) UpdateCommand(parserCommand *schema.Command) error {
+
+	var errorString []string
+	testCommand, found := getSchemaCommand(devfile.SchemaDevFile.Commands, parserCommand.Id)
+	if found {
+		LogInfoMessage(fmt.Sprintf("Updating command id: %s", parserCommand.Id))
+		if testCommand.Exec != nil {
+			setExecCommandValues(parserCommand.Exec)
+		} else if testCommand.Composite != nil {
+			setCompositeCommandValues(parserCommand.Composite)
+		}
+		devfile.replaceSchemaCommand(*parserCommand)
+	} else {
+		errorString = append(errorString, LogErrorMessage(fmt.Sprintf("Command not found in test : %s", parserCommand.Id)))
+	}
+
+	var returnError error
+	if len(errorString) > 0 {
+		returnError = errors.New(fmt.Sprint(errorString))
+	}
+	return returnError
+}
+
+// Create and return an exec command in a schema structure
+func createExecCommand() *schema.ExecCommand {
+
+	LogInfoMessage("Create an exec command :")
+	execCommand := schema.ExecCommand{}
+	setExecCommandValues(&execCommand)
+	return &execCommand
+
+}
+
+// Set the attribute values of an exec command
+func setExecCommandValues(execCommand *schema.ExecCommand) {
+
+	execCommand.Component = GetRandomString(8, false)
+	LogInfoMessage(fmt.Sprintf("....... component: %s", execCommand.Component))
+
+	execCommand.CommandLine = GetRandomString(4, false) + " " + GetRandomString(4, false)
+	LogInfoMessage(fmt.Sprintf("....... commandLine: %s", execCommand.CommandLine))
+
+	if GetRandomDecision(2, 1) {
+		execCommand.Group = addGroup()
+	} else {
+		execCommand.Group = nil
+	}
+
+	if GetBinaryDecision() {
+		execCommand.Label = GetRandomString(12, false)
+		LogInfoMessage(fmt.Sprintf("....... label: %s", execCommand.Label))
+	} else {
+		execCommand.Label = ""
+	}
+
+	if GetBinaryDecision() {
+		execCommand.WorkingDir = "./tmp"
+		LogInfoMessage(fmt.Sprintf("....... WorkingDir: %s", execCommand.WorkingDir))
+	} else {
+		execCommand.WorkingDir = ""
+	}
+
+	execCommand.HotReloadCapable = GetBinaryDecision()
+	LogInfoMessage(fmt.Sprintf("....... HotReloadCapable: %t", execCommand.HotReloadCapable))
+
+	if GetBinaryDecision() {
+		execCommand.Env = AddEnv(GetRandomNumber(4))
+	} else {
+		execCommand.Env = nil
+	}
+
+}
+
+// Use the specified command to replace the command in the schema structure with the same Id.
+func (devfile TestDevfile) replaceSchemaCommand(command schema.Command) {
+	for i := 0; i < len(devfile.SchemaDevFile.Commands); i++ {
+		if devfile.SchemaDevFile.Commands[i].Id == command.Id {
+			devfile.SchemaDevFile.Commands[i] = command
+			break
+		}
+	}
+}
```

Comment:
I think if we clean up the test devfile structs like I mentioned earlier you can use the parser's updateCommand() func. This is basically duplicating effort and having two copies of the same thing - one in test and one in parser

Reply:
Yes, just the way the test work, maintain a reference of devfile content to compare with what the parser returns.